rust/trace/src/chrome_trace_dump.rs

```diff
@@ -82,6 +83,36 @@ where
     serde_json::from_reader(input).map_err(Error::Json)
 }
 
+/// Save tracing data to to supplied path, using the Trace Viewer format. Trace file can be opened
+/// using the Chrome browser by visiting the URL `about:tracing`. If `sorted_chronologically` is
+/// true then sort output traces chronologically by each trace's time of creation.
+pub fn save_traces(output_path: &Path, sorted_chronologically: bool) -> IOResult<()> {
+    let traces = if sorted_chronologically {
+        super::samples_cloned_sorted()
+    } else {
+        super::samples_cloned_unsorted()
+    };
+
+    match File::create(&output_path) {
+        Ok(f) => f,
+        Err(err) => {
+            eprintln!("Could not create trace output file at: {}.", &output_path);
+            return IOResult(err);
+        }
+    };
+
+    match serialize(&traces, &mut trace_file) {
```

Comment:
ooh, I didn't realize that this has a different result type. Luckily we actually already have a custom Error type in this crate, so we should be able to just use that as our return type. In this case as well we can let the caller handle the error, and we don't need to do any printing here.